packages/connectkit/src/components/DaimoPayButton/index.tsx

```diff
@@ -1,128 +1,217 @@
 
-const contentVariants: Variants = {
-  initial: {
-    zIndex: 2,
-    opacity: 0,
-    x: "-100%",
-  },
-  animate: {
-    opacity: 1,
-    x: 0.1,
-    transition: {
-      duration: 0.4,
-      ease: [0.25, 1, 0.5, 1],
-    },
-  },
-  exit: {
-    zIndex: 1,
-    opacity: 0,
-    x: "-100%",
-    pointerEvents: "none",
-    position: "absolute",
-    transition: {
-      duration: 0.4,
-      ease: [0.25, 1, 0.5, 1],
-    },
-  },
-};
+type PayButtonCommonProps =
+  | {
+      /**
+       * Your public app ID. Specify either (payId) or (appId + parameters).
+       */
+      appId: string;
+      /**
+       * Optional nonce. If set, generates a deterministic payID. See docs.
+       */
+      nonce?: bigint;
+      /**
+       * Destination chain ID.
+       */
+      toChain: number;
+      /**
+       * The destination token to send, completing payment. Must be an ERC-20
+       * token or the zero address, indicating the native token / ETH.
+       */
+      toToken: Address;
+      /**
+       * The amount of destination token to send (transfer or approve).
+       */
+      toAmount: bigint;
+      /**
+       * The destination address to transfer to, or contract to call.
+       */
+      toAddress: Address;
+      /**
+       * Optional calldata to call an arbitrary function on `toAddress`.
+       */
+      toCallData?: Hex;
+      /**
+       * The intent verb, such as "Pay", "Deposit", or "Purchase".
+       */
+      intent?: string;
+      /**
+       * Payment options. By default, all are enabled.
+       */
+      paymentOptions?: PaymentOption[];
+      /**
+       * Preferred chain IDs. Assets on these chains will appear first.
+       */
+      preferredChains?: number[];
+    }
+  | {
+      /** The payment ID, generated via the Daimo Pay API. Replaces params above. */
+      payId: string;
+    };
 
-type Hash = `0x${string}`;
+type DaimoPayButtonProps = PayButtonCommonProps & {
+  /** Light mode, dark mode, or auto. */
+  mode?: Mode;
+  /** Named theme. See docs for options. */
+  theme?: Theme;
+  /** Custom theme. See docs for options. */
+  customTheme?: CustomTheme;
+  /** Automatically close the modal after a successful payment. */
+  closeOnSuccess?: boolean;
+  /** Get notified when the user clicks, opening the payment modal. */
+  onClick?: (open: () => void) => void;
+};
 
-type DaimoPayButtonRendererProps = {
-  /** The payment ID, generated via the Daimo Pay API. */
-  payId: string;
+type DaimoPayButtonCustomProps = PayButtonCommonProps & {
   /** Automatically close the modal after a successful payment. */
   closeOnSuccess?: boolean;
   /** Custom renderer */
-  children?: (renderProps: {
-    show?: () => void;
-    hide?: () => void;
-    chain?: Chain & {
-      unsupported?: boolean;
-    };
-    unsupported: boolean;
-    isConnected: boolean;
-    isConnecting: boolean;
-    address?: Hash;
-    truncatedAddress?: string;
-    ensName?: string;
-    payId?: string;
```

Comment:
I think we need this if we put text in the button from the payment

Reply:
with the new style, the payment isn't generated till hydration. we do generate what the payId will be, but might be confusing to pass thru `payId` that doesn't exist when looked up in API

can pass thru intent, but this is the client's own page, right? say they make a custom Donate button + make the payment with intent=Donate so modal shows nicely